pkg/cmd/server/origin/handlers.go

```diff
@@ -168,6 +176,98 @@ func namespacingFilter(handler http.Handler, contextMapper kapi.RequestContextMa
 	})
 }
 
+// versionSkewFilter adds a filter that may deny PUT an POST requests since they stomp entire resource objects from skewed
+// oc clients, since we know that those clients will strip unknown fields which can lead to unexpected outcomes
+func (c *MasterConfig) versionSkewFilter(binaryInfo version.Info, handler http.Handler) http.Handler {
+	skewedClientPolicy := c.Options.PolicyConfig.SkewedClientPolicy
+	if skewedClientPolicy == configapi.AllowAll {
+		return handler
+	}
+	seg := strings.SplitN(binaryInfo.GitVersion, "-", 2)
+	serverVersionNumber := seg[0][1:]
+	serverCommit := binaryInfo.GitCommit
+	if len(serverCommit) > 7 {
+		serverCommit = serverCommit[:7]
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if req.Method != "PUT" && req.Method != "POST" {
```

Comment:
No, skewed usage checks are not only useful for stripping logic, they are
primarily used by admins to force clients to upgrade because the logic the
client is using in the patch is faulty.  The *goal* of skew checks is to
deny clients.  *sometimes* the reason a client is denied is because it
doesn't follow correct logic.

On Wed, Mar 2, 2016 at 5:10 PM, David Eads <notifications@github.com> wrote:

> In pkg/cmd/server/origin/handlers.go
> <https://github.com/openshift/origin/pull/7741#discussion_r54809142>:
>
> > +// versionSkewFilter adds a filter that may deny PUT an POST requests since they stomp entire resource objects from skewed
> > +// oc clients, since we know that those clients will strip unknown fields which can lead to unexpected outcomes
> > +func (c *MasterConfig) versionSkewFilter(binaryInfo version.Info, handler http.Handler) http.Handler {
> > +	skewedClientPolicy := c.Options.PolicyConfig.SkewedClientPolicy
> > +	if skewedClientPolicy == configapi.AllowAll {
> > +		return handler
> > +	}
> > +	seg := strings.SplitN(binaryInfo.GitVersion, "-", 2)
> > +	serverVersionNumber := seg[0][1:]
> > +	serverCommit := binaryInfo.GitCommit
> > +	if len(serverCommit) > 7 {
> > +		serverCommit = serverCommit[:7]
> > +	}
> > +
> > +	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
> > +		if req.Method != "PUT" && req.Method != "POST" {
>
> Not patch. Patches are non-encoded json that avoids any field stripping
> logic. That stripping logic is the only reason to block skewed usage.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7741/files#r54809142>.
>